cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -714,7 +715,81 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
                 AbstractVisibilityEditor.prototype.getContext.call(this),
                 {
                     hide_after_due: this.modelVisibility() === 'hide_after_due',
-                    self_paced: this.model.get('self_paced') === true
+                    self_paced: course.get('self_paced') === true
+                }
+            );
+        }
+    });
+
+    ShowCorrectnessEditor = AbstractEditor.extend({
+        templateName: 'show-correctness-editor',
+        className: 'edit-show-correctness',
+        events: {
+            'change input[name=show-correctness]': 'togglePastDueWarning'
+        },
+
+        modelValue: function() {
+            var showCorrectness = this.model.get('show_correctness');
+            if (showCorrectness === 'always' ||
```

Comment:
In the test universe, it can be null/undefined, but you're right, there's cleaner ways to handle this.  Have fixed with https://github.com/edx/edx-platform/pull/14737/commits/541b3c9d4d39e3d84910ea242c86c9dc4569e069#diff-fecac9c1a5c871b71bafab46e808b363L743.